assets/js/base/components/shipping-location/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	SHIPPING_COUNTRIES,
+	SHIPPING_STATES,
+} from '@woocommerce/block-settings';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Shows a formatted shipping location.
+ */
+const ShippingLocation = ( { address } ) => {
+	const formattedCountry =
+		typeof SHIPPING_COUNTRIES[ address.country ] === 'string'
+			? decodeEntities( SHIPPING_COUNTRIES[ address.country ] )
+			: '';
+
+	const formattedState =
+		typeof SHIPPING_STATES[ address.country ] === 'object' &&
+		typeof SHIPPING_STATES[ address.country ][ address.state ] === 'string'
+			? decodeEntities(
+					SHIPPING_STATES[ address.country ][ address.state ]
+			  )
+			: address.state;
+
+	const addressParts = [];
+
+	addressParts.push( address.postcode.toUpperCase() );
+	addressParts.push( address.city );
+	addressParts.push( formattedState );
+	addressParts.push( formattedCountry );
+
+	const formattedLocation = addressParts.filter( Boolean ).join( ', ' );
```

Comment:
What about RTL languages? Doesn't this force LTR?

Reply:
Interesting point, I don't think we need to give any special treatment to RTL languages here, the order will be the same (1st postcode, 2nd city, 3rd state, 4th country) and the browser will take care of displaying the commas (`, `) LTR or RTL depending on the direction of the text they are displayed within.

I found some weird results when mixing LTR and RTL languages in the same address (for example, using RTL for all fields but LTR for postal codes), but I don't know how common that is. :thinking: 

Maybe @senadir can give a hand here?

Reply:
RTL should not have special treatment at all in strings, the address will show correctly as long as the page is RTL, problems can exist (zipcode being last) if you're in a LTR page.